www/js/diary/services.js

```diff
@@ -412,10 +412,20 @@ angular.module('emission.main.diary.services', ['emission.plugin.logger',
               return UnifiedDataLoader.getUnifiedMessagesForInterval(
                   inp_key, pendingLabelQuery).then(manualFactory.extractResult);
             });
+            var enbsPromises = enbs.MANUAL_KEYS.map(function(inp_key) {
+              return UnifiedDataLoader.getUnifiedMessagesForInterval(
+                  inp_key, pendingLabelQuery).then(enbs.extractResult);
+            });
             const manualConfirmResults = {};
-            return Promise.all(manualPromises).then((manualResults) => {
-                manualFactory.processManualInputs(manualResults, manualConfirmResults);
-                return [result, manualConfirmResults];
+            const enbsConfirmResults = {};
+            return new Promise((rs, rj) => {
+                Promise.all(manualPromises).then(function(results) {
+                  manualFactory.processManualInputs(results, manualConfirmResults);
+                });
+                Promise.all(enbsPromises).then(function(results) {
+                  enbs.processManualInputs(results, enbsConfirmResults);
+                });
+                rs([result, manualConfirmResults, enbsConfirmResults]);
```

Comment:
I'm not sure this actually works - in the old code, we would only return after all the `manualPromises` had completed (return is within the `then`, but here, we don't wait for `manualPromises` or `enbsPromises` before we call `rs`.

I think you need

```
return Promise.all(manualPromises.concat(enbsPromises)).then([comboResults] => {
      manualResults = comboResults.slice(0,manualPromises.length);
...
      return [result, manualConfirmResults, enbsConfirmResults];
});
```

Reply:
You're absolutely right. I'll implement this with spread syntax